docs/source/Objective_Functions/Objective_Function.md

```diff
@@ -1,25 +1,248 @@
 [2]: https://en.wikipedia.org/wiki/Pareto_front
 
-### Objective Function Format
-A TQ42 optimization algorithm must be provided with one or more real-valued functions to be optimized. The objective function(s) must be provided in the format of an endpoint accessible through https.
+## Objective Function and Local Optimization Function Format
+A TQ42 optimization algorithm must be provided with one or more real-valued functions to be optimized. The objective and local optimization function must be provided in the format of:
+1. A communication channel
+2. An https endpoint
 
+
+An example of TetraOpt parameters using the `communication channel`. Notice the
+`objective_function_channel_id` and `local_optimizer_channel_id parameter`:
+
+```
+objective_func_channel = await Channel.create(client=client)
+local_opt_channel = await Channel.create(client=client)
+
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+     "objective_function_channel_id": objective_func_channel
+     #local_optimizer_channel_id parameter is optional
+    "local_optimizer_channel_id": local_opt_channel
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+An example of TetraOpt parameters using `an https endpoint`. Notice the
+`objective_function` and `local_optimizer parameter`:
+```
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+    'objective_function':'http://127.0.0.1:8000/test_func_eval/Ackley/',
+     #local_optimizer parameter is optional  
+    "local_optimizer": "http://127.0.0.1:8000/local_optimization/Ackley/",
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+### 1. To use the communications channel:
+The channel uses objects of the following classes to send and receive information:
+1. `Ask`
+2. `Tell`
+
+It also need the following class to stream the information to TetraOpt
+1. `Channel`
+
+We can import them from the tq42 library in our Python script using the following:
+1. `from tq42.channel import Channel, Ask, Tell`
+
+
+#### The `Ask` Class
+In our example above, we used the Ackley function provided by the OptimizationTestFunctions library as our objective function. 
+
+An Example of an `Ask` object values passed by TetraOpt:
+```
+{
+    parameters {
+      values: 1
+      values: 3
+    }
+    parameters {
+      values: 2
+      values: 7
+    }
+    headers: "h0"
+    headers: "h1"
+}
+```
+
+The fields are formally described in Google Protocol Buffers as:
+```
+message Ask {
```

Comment:
I am not a fan of referencing a protobuf schema here. If you ever update the protobuf schema this gets outdated quickly. I would advise to give an example at most. Python types should already guide the user.